src/Promitor.Agents.ResourceDiscovery/Validation/Steps/AzureLandscapeValidationStep.cs

```diff
@@ -54,5 +60,51 @@ public ValidationResult Run()
 
             return errorMessages.Any() ? ValidationResult.Failure(ComponentName, errorMessages) : ValidationResult.Successful(ComponentName);
         }
+
+        private IEnumerable<string> ValidateAzureCloudEnvironment()
```

Comment:
We can reduce this to just validate _azureLandscape.AzureEnvironment's properties when it is not null. In that case they should all be provided